### PR TITLE
[dhctl] Preflight check master hostname

### PR DESCRIPTION
**Pull request by aleksey-su**
## Description
Preflight check hostname uniqueness on servers when setting up a bare metal cluster and check only one --ssh-host parameter used

## Why do we need it, and what problem does it solve?
Before performing bootstrap, checks hostname uniqueness on servers when setting up a bare metal cluster and checks only one --ssh-host parameter used

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: dhctl
type: feature
summary: Preflight check hostname uniqueness on servers when setting up a bare metal cluster and check only one --ssh-host parameter used
impact_level: default
```

### CODE REVIEW

**Comment by aleksey-su** [comment]
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Comment by aleksey-su** [comment]
/e2e/abort openstack;Standard;containerd;1.25 9286335234 9286335234-1-con-1-25 /sys/deckhouse-oss/install:pr8515

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Comment by aleksey-su** [comment]
/e2e/abort openstack;Standard;containerd;1.25 9286587856 9286587856-2-con-1-25 /sys/deckhouse-oss/install:pr8515

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Comment by aleksey-su** [comment]
/e2e/abort static;Static;containerd;1.25 9345350760 9345350760-1-con-1-25 /sys/deckhouse-oss/install:pr8515 astra@192.168.199.17

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Comment by aleksey-su** [comment]
/e2e/abort static;Static;containerd;1.25 9345598302 9345598302-1-con-1-25 /sys/deckhouse-oss/install:pr8515 astra@192.168.199.170

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by aleksey-su** [comment]
image

**Comment by aleksey-su** [comment]
image
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Comment by aleksey-su** [comment]
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Comment by aleksey-su** [comment]
/e2e/abort static;Static;containerd;1.25 9548604724 9548604724-1-con-1-25 /sys/deckhouse-oss/install:pr8515 astra@192.168.199.136

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Comment by aleksey-su** [comment]
image
image

**Comment by aleksey-su** [comment]
image
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Comment by aleksey-su** [comment]
/e2e/abort static;Static;containerd;1.27 9577624365 9577624365-1-con-1-27 /sys/deckhouse-oss/install:pr8515 astra@192.168.199.203

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

File: dhctl/pkg/app/preflight.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2023 Flant JSC
+// Copyright 2024 Flant JSC
```
Comment (mvasl) [inline dhctl/pkg/app/preflight.go:1]: No need for this, file was first created in 2023

File: dhctl/pkg/preflight/preflight.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2023 Flant JSC
+// Copyright 2024 Flant JSC
```
Comment (mvasl) [inline dhctl/pkg/preflight/preflight.go:1]: No need for this, file was first created in 2023

File: dhctl/pkg/preflight/ssh.go [hunk 3]
```diff
@@ -64,7 +68,24 @@ func (pc *Checker) CheckSSHCredential() error {
...
+
+func (pc *Checker) CheckSingleSSHHostForStatic() error {
+	if app.PreflightSkipOneSSHHost {
+		log.InfoLn("Only one --ssh-host parameter used preflight check was skipped")
+		return nil
+	}
+
+	if len(pc.sshClient.Settings.AvailableHosts()) > 1 {
+		return fmt.Errorf(
+			"for the bootstrap first master, only one --ssh-host parameter should be used for a static cluster",
```
Comment (mvasl) [inline dhctl/pkg/preflight/ssh.go:87]: "During the bootstrap of the first static master node, only one --ssh-host parameter is allowed"

File: dhctl/pkg/template/preflight.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2023 Flant JSC
+// Copyright 2024 Flant JSC
```
Comment (mvasl) [inline dhctl/pkg/template/preflight.go:1]: No need for this, file was first created in 2023

**Review by mvasl — CHANGES_REQUESTED** [review]

**Comment by aleksey-su** [comment]
/e2e/abort static;Static;containerd;1.27 9611766608 9611766608-1-con-1-27 /sys/deckhouse-oss/install:pr8515 astra@192.168.199.144

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.27 9552600575 9552600575-1-con-1-27 /sys/deckhouse-oss/install:pr8515 astra@192.168.199.188

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Comment by mvasl** [comment]
Is there really no way to make candi/bashible/common-steps/node-group/069_check_hostname_uniqueness.sh.tpl cleaner? It's really hard to read and will be a nightmare to support

**Review by name212 — APPROVED** [review]

**Review by mvasl — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @aleksey-su. I've started the workflow for label 'e2e/run/static'...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.27 9889209892 9889209892-2-con-1-27 /sys/deckhouse-oss/install:pr8515 astra@192.168.199.241

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.27 9889209892 9889209892-1-con-1-27 /sys/deckhouse-oss/install:pr8515 astra@192.168.199.5

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Review by RomanenkoDenys — APPROVED** [review]

**Pull request merged**